packages/core/src/node/env-variables/env-variables-server.ts

```diff
@@ -44,4 +52,24 @@ export class EnvVariablesServerImpl implements EnvVariablesServer {
         }
         return this.envs[key];
     }
+
+    async getUserHomeFolderPath(): Promise<string> {
+        return os.homedir();
+    }
+
+    async getDataFolderName(): Promise<string> {
+        return THEIA_DATA_FOLDER;
+    }
+
+    async getUserDataFolderPath(): Promise<string> {
+        return path.join(os.homedir(), THEIA_DATA_FOLDER);
+    }
+
+    async getAppDataPath(): Promise<string> {
+        return path.join(
+            os.homedir(),
```

Comment:
it should delegate to `getUserHomeFolderPath` and `getDataFolderName`, that if they are overriden then `getAppDataPath` respects it. The same for `getUserDataFolderPath`.

Reply:
Agree, if there is an override for `getDataFolderName` other methods should respect it.